### DISCUSSION TITLE
Is there a way to set the useAsTitle property with a relationship's field?

### DISCUSSION THREAD

**Discussion by fvcci** [payloadcms/payload#3304] [Feature Requests & Ideas]
Let's say I have a collection named "Person" and another collection named "Cyclist". They follow an Entity-relationship model where Cyclist has the properties of a Person like their "name" for example. Cyclist has the relationship field to Person, and I want to set the useAsTitle field when creating a Cyclist to the Person's name. Is this possible?

**Comment by matthijs166** [comment]
Maybe you can do something like this:

```js
 {
            name: "title",
            type: "text",
            hooks: {
                beforeChange: [
                    async ({ data }) => {
                        // fallback?
                        // also dont know if data.cyclistRelation is an ID or the full object
                        // otherwise use the internal payload api here
                        return `$(data.name) - $(data.cyclistRelation.title)`
                    }
                ]
            }
        },
```

**Reply by fvcci** [reply]
data.cyclistRelation is an ID. I'm not sure what you mean when you are referring to the intern payload api. Could you explain?

**Comment by GeorgeHulpoi** [comment]
Is related to https://github.com/payloadcms/payload/discussions/3257

I think it can be more useful than using the virtual fields paradigm.
